src/Utilities/SingleStackUtils/SingleStackUtils.jl

```diff
@@ -204,4 +208,108 @@ function get_horizontal_variance(
     return vars_var
 end
 
+"""
+    reduce_nodal_stack(
+        op::Function,
+        grid::DiscontinuousSpectralElementGrid{T, dim, N},
+        Q::MPIStateArray,
+        vars::NamedTuple,
+        var::String;
+        vrange::UnitRange = 1:size(Q, 3),
+    ) where {T, dim, N}
+
+Reduce `var` from `vars` within `Q` over all nodal points in the specified
+`vrange` of elements with `op`. Return a tuple `(result, z)` where `result` is
+the final value returned by `op` and `z` is the index within `vrange` where the
+`result` was determined.
+"""
+function reduce_nodal_stack(
+    op::Function,
+    grid::DiscontinuousSpectralElementGrid{T, dim, N},
+    Q::MPIStateArray,
+    vars::Type,
+    var::String;
+    vrange::UnitRange = 1:size(Q, 3),
+    i::Int = 1,
+    j::Int = 1,
+) where {T, dim, N}
+    Nq = N + 1
+    Nqk = dimensionality(grid) == 2 ? 1 : Nq
+
+    var_names = flattenednames(vars)
+    var_ind = findfirst(s -> s == var, var_names)
+    if var_ind === nothing
+        return
+    end
+
+    state_data = array_device(Q) isa CPU ? Q.realdata : Array(Q.realdata)
+    z = vrange.start
+    result = state_data[1, var_ind, z]
+    for ev in vrange
+        for k in 1:Nqk
+            ijk = i + Nq * ((j - 1) + Nq * (k - 1))
+            new_result = op(result, state_data[ijk, var_ind, ev])
+            if !isequal(new_result, result)
+                result = new_result
+                z = ev
+            end
+        end
+    end
+
+    return (result, z)
+end
+
+"""
+    reduce_element_stack(
+        op::Function,
+        grid::DiscontinuousSpectralElementGrid{T, dim, N},
+        Q::MPIStateArray,
+        vars::NamedTuple,
+        var::String;
+        vrange::UnitRange = 1:size(Q, 3),
+    ) where {T, dim, N}
+
+Reduce `var` from `vars` within `Q` over all nodal points in the specified
+`vrange` of elements with `op`. Return a tuple `(result, z)` where `result` is
+the final value returned by `op` and `z` is the index within `vrange` where the
+`result` was determined.
+"""
+function reduce_element_stack(
+    op::Function,
+    grid::DiscontinuousSpectralElementGrid{T, dim, N},
+    Q::MPIStateArray,
+    vars::Type,
+    var::String;
+    vrange::UnitRange = 1:size(Q, 3),
+) where {T, dim, N}
+    Nq = N + 1
+    nodal_stacks = [
```

Comment:
I think we just need
```julia
    return [
```

This way the user can decide how to parse the nodal results.